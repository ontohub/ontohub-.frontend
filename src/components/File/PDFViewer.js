import React from "react";
import styled, { css } from "styled-components";
import { compose, defaultProps, withProps, withStateHandlers } from "recompose";
import { Button, Icon, Label } from "semantic-ui-react";
import PDF from "react-pdf-js";

const RawPagination = ({
  className,
  page,
  pagesCount,
  handlePrevious,
  handleNext
}) => {
  return (
    <Button.Group className={className}>
      <Button
        onClick={handlePrevious}
        disabled={page === 1}
        icon
        labelPosition="left"
      >
        <Icon name="angle double left" />
        Previous Page
      </Button>
      <Button.Or text={`${page}/${pagesCount}`} />
      <Button
        onClick={handleNext}
        disabled={page === pagesCount}
        icon
        labelPosition="right"
      >
        Next Page
        <Icon name="angle double right" />
      </Button>
    </Button.Group>
  );
};
const Pagination = styled(RawPagination)`
  text-align: center;
  width: 100%;
`;

const Viewer = ({
  page,
  pagesCount,
  handlePrevious,
  handleNext,
  onDocumentComplete,
  onPageComplete,
  ...pdfProps
}) => (
  <div>
    <Pagination
      page={page}
      pagesCount={pagesCount}
      handlePrevious={handlePrevious}
      handleNext={handleNext}
    />
    <PDF
      onDocumentComplete={onDocumentComplete}
      onPageComplete={onPageComplete}
      page={page}
      {...pdfProps}
    />
  </div>
);

export default compose(
  defaultProps({ fillWidth: true, fillHeight: true }),
  withStateHandlers(
    ({ initialPage = 1, initialPagesCount = 1 }) => ({
      page: initialPage,
      pagesCount: initialPagesCount
    }),
    {
      onDocumentComplete: () => pagesCount => ({
        page: 1,
        pagesCount: pagesCount
      }),
      onPageComplete: () => page => ({
        page: page
      }),
      handlePrevious: ({ page }) => () => ({
        page: page - 1
      }),
      handleNext: ({ page }) => () => ({
        page: page + 1
      })
    }
  )
)(Viewer);
