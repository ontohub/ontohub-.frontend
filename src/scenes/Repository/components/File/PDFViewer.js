import React from "react";
import styled from "styled-components";
import { compose, defaultProps, withStateHandlers } from "recompose";
import { Button, Icon, Input, Label } from "semantic-ui-react";
import PDF from "react-pdf-js";

const UnstyledPagination = ({
  className,
  page,
  pageInInput,
  pagesCount,
  handlePrevious,
  handleNext,
  changePageInInputTo,
  changePageTo
}) => {
  return (
    <div className={`row ${className}`}>
      <Button
        onClick={handlePrevious}
        disabled={page === 1}
        icon
        labelPosition="left"
      >
        <Icon name="angle double left" />
        Previous Page
      </Button>
      <form
        onSubmit={event => {
          event.preventDefault();
          changePageTo(pageInInput);
        }}
      >
        <Input labelPosition="right" type="text" placeholder="Page">
          <input
            style={{ textAlign: "right" }}
            value={pageInInput}
            onChange={event => {
              changePageInInputTo(event.target.value);
            }}
          />
          <Label>of {pagesCount}</Label>
        </Input>
      </form>
      <Button
        onClick={handleNext}
        disabled={page === pagesCount}
        icon
        labelPosition="right"
      >
        Next Page
        <Icon name="angle double right" />
      </Button>
    </div>
  );
};
const Pagination = styled(UnstyledPagination)`
  text-align: center;
  width: 100%;
  & form {
    display: inline-block;
    margin: 0 0.25em 0 0;
    padding: 0.125em 0 0 0;
  }
  & form input {
    width: 4em;
    height: 36px;
  }
  & form .label {
    width: 4em;
    height: 36px;
    text-align: left;
  }
`;

const PurePDFViewer = ({
  page,
  pagesCount,
  pageInInput,
  handlePrevious,
  handleNext,
  changePageInInputTo,
  changePageTo,
  onDocumentComplete,
  onPageComplete,
  ...pdfProps
}) => (
  <div>
    <Pagination
      page={page}
      pageInInput={pageInInput}
      pagesCount={pagesCount}
      handlePrevious={handlePrevious}
      handleNext={handleNext}
      changePageTo={changePageTo}
      changePageInInputTo={changePageInInputTo}
    />
    <PDF
      onDocumentComplete={onDocumentComplete}
      onPageComplete={onPageComplete}
      page={page}
      {...pdfProps}
    />
  </div>
);

export const PDFViewer = compose(
  defaultProps({ fillWidth: true, fillHeight: true }),
  withStateHandlers(
    ({ initialPage = 1, initialPagesCount = 1 }) => ({
      page: initialPage,
      pagesCount: initialPagesCount,
      pageInInput: initialPage
    }),
    {
      onDocumentComplete: () => pagesCount => ({
        page: 1,
        pagesCount: pagesCount
      }),
      onPageComplete: () => page => ({
        page: page,
        pageInInput: page
      }),
      handlePrevious: ({ page }) => () => ({
        page: page - 1
      }),
      handleNext: ({ page }) => () => ({
        page: page + 1
      }),
      changePageInInputTo: () => newPage => ({
        pageInInput: newPage
      }),
      changePageTo: ({ pagesCount }) => newPage => {
        const pageInt = parseInt(newPage, 10);
        if (pageInt && 1 <= pageInt && pageInt <= pagesCount) {
          return { page: pageInt };
        }
      }
    }
  )
)(PurePDFViewer);
