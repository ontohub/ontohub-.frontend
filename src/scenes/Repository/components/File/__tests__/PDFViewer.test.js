import { PDFViewer } from "../PDFViewer";

describe("PDFViewer", () => {
  set("component", () => <PDFViewer />);
  set("renderedComponent", () => render(component));

  it("matches the snapshot", () => {
    expect(renderedComponent).toMatchSnapshot();
  });
});
