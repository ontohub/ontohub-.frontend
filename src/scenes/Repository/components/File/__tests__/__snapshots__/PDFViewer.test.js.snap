// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PDFViewer matches the snapshot 1`] = `
.c0 {
  text-align: center;
  width: 100%;
}

.c0 form {
  display: inline-block;
  margin: 0 0.25em 0 0;
  padding: 0.125em 0 0 0;
}

.c0 form input {
  width: 4em;
  height: 36px;
}

.c0 form .label {
  width: 4em;
  height: 36px;
  text-align: left;
}

<div>
  <div
    class="row c0"
  >
    <button
      class="ui icon disabled left labeled button"
      disabled=""
      role="button"
      tabindex="-1"
    >
      <i
        aria-hidden="true"
        class="angle double left icon"
      >
        
      </i>
      Previous Page
    </button>
    <form>
      <div
        class="ui right labeled input"
      >
        <input
          placeholder="Page"
          style="text-align:right"
          type="text"
          value="1"
        >
          
        </input>
        <div
          class="ui label"
        >
          of 1
        </div>
      </div>
    </form>
    <button
      class="ui icon disabled right labeled button"
      disabled=""
      role="button"
      tabindex="-1"
    >
      Next Page
      <i
        aria-hidden="true"
        class="angle double right icon"
      >
        
      </i>
    </button>
  </div>
  <div>
    Loading PDF...
  </div>
</div>
`;
